pallets/collator-selection/src/lib.rs

```diff
@@ -555,6 +578,126 @@ pub mod pallet {
 			Self::deposit_event(Event::InvulnerableRemoved { account_id: who });
 			Ok(())
 		}
+
+		/// Todo
+		///
+		/// Todo
+		#[pallet::call_index(7)]
+		#[pallet::weight(T::WeightInfo::increase_bond(T::MaxCandidates::get()))]
+		pub fn increase_bond(origin: OriginFor<T>, bond: BalanceOf<T>) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			let new_bond =
+				<Candidates<T>>::try_mutate(|candidates| -> Result<BalanceOf<T>, DispatchError> {
+					let candidate_info = candidates
+						.iter_mut()
+						.find(|candidate_info| candidate_info.who == who)
+						.ok_or_else(|| Error::<T>::NotCandidate)?;
+					candidate_info.deposit = candidate_info.deposit.saturating_add(bond);
+					T::Currency::reserve(&who, bond)?;
+					T::CandidateList::on_increase(&who, bond)
+						.map_err(|_| Error::<T>::OnIncrease)?;
+					Ok(candidate_info.deposit)
+				})?;
+
+			Self::deposit_event(Event::CandidateBondIncreased {
+				account_id: who,
+				deposit: new_bond,
+			});
+			Ok(())
+		}
+
+		/// Todo
+		///
+		/// Todo
+		#[pallet::call_index(8)]
+		#[pallet::weight(T::WeightInfo::decrease_bond(T::MaxCandidates::get()))]
+		pub fn decrease_bond(origin: OriginFor<T>, bond: BalanceOf<T>) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			let new_bond =
+				<Candidates<T>>::try_mutate(|candidates| -> Result<BalanceOf<T>, DispatchError> {
```

Comment:
I have a suggestion for you here:  the implementation of the increase and decrease bond extrinsics are very similar. I'd suggest refactoring out most it it's mutation logic into a `Self::update_bond(accountID, imbalance)`, where the imbalance is an enum that can take positive and negative imbalance (to add or remove bond).

The pallet balances implements a similar logic and, perhaps more relevant, I've implemented something similar in this PR, which also calls into a `SortedListProvoder::on_increase/decrease` https://github.com/paritytech/substrate/pull/14620

Reply:
I think it's a good idea.